From: Tony Jones <tonyj@suse.de>
Subject: drop additional perfmon2 syscalls
Patch-mainline: never

Patch accepted from SGI (bnc#430298, FATE #303968) added 12 new syscalls
which have been replaced (for SLES11) with an ioctl based interface.

Remove the relevant syscalls.

Signed-off-by: Tony Jones <tonyj@suse.de>

---

 arch/ia64/include/asm/unistd.h     |   14 +-------------
 arch/ia64/kernel/entry.S           |   12 ------------
 arch/powerpc/include/asm/systbl.h  |   12 ------------
 arch/powerpc/include/asm/unistd.h  |   14 +-------------
 arch/x86/ia32/ia32entry.S          |   12 ------------
 arch/x86/include/asm/ia32_unistd.h |   13 ++++++-------
 arch/x86/include/asm/unistd_32.h   |   14 --------------
 arch/x86/include/asm/unistd_64.h   |   25 +------------------------
 arch/x86/kernel/syscall_table_32.S |   12 ------------
 kernel/sys_ni.c                    |   13 -------------
 10 files changed, 9 insertions(+), 132 deletions(-)

--- a/arch/ia64/include/asm/unistd.h
+++ b/arch/ia64/include/asm/unistd.h
@@ -308,23 +308,11 @@
 #define __NR_dup3			1316
 #define __NR_pipe2			1317
 #define __NR_inotify_init1		1318
-#define __NR_pfm_create_context		1319
-#define __NR_pfm_write_pmcs		(__NR_pfm_create_context+1)
-#define __NR_pfm_write_pmds		(__NR_pfm_create_context+2)
-#define __NR_pfm_read_pmds		(__NR_pfm_create_context+3)
-#define __NR_pfm_load_context		(__NR_pfm_create_context+4)
-#define __NR_pfm_start			(__NR_pfm_create_context+5)
-#define __NR_pfm_stop			(__NR_pfm_create_context+6)
-#define __NR_pfm_restart		(__NR_pfm_create_context+7)
-#define __NR_pfm_create_evtsets		(__NR_pfm_create_context+8)
-#define __NR_pfm_getinfo_evtsets 	(__NR_pfm_create_context+9)
-#define __NR_pfm_delete_evtsets 	(__NR_pfm_create_context+10)
-#define __NR_pfm_unload_context		(__NR_pfm_create_context+11)
 
 #ifdef __KERNEL__
 
 
-#define NR_syscalls			307 /* length of syscall table */
+#define NR_syscalls			295 /* length of syscall table */
 
 /*
  * The following defines stop scripts/checksyscalls.sh from complaining about
--- a/arch/ia64/kernel/entry.S
+++ b/arch/ia64/kernel/entry.S
@@ -1703,18 +1703,6 @@ sys_call_table:
 	data8 sys_dup3
 	data8 sys_pipe2
 	data8 sys_inotify_init1
-	data8 sys_pfm_create_context
-	data8 sys_pfm_write_pmcs		// 1320
-	data8 sys_pfm_write_pmds
-	data8 sys_pfm_read_pmds
-	data8 sys_pfm_load_context
-	data8 sys_pfm_start
-	data8 sys_pfm_stop			// 1325
-	data8 sys_pfm_restart
-	data8 sys_pfm_create_evtsets
-	data8 sys_pfm_getinfo_evtsets
-	data8 sys_pfm_delete_evtsets
-	data8 sys_pfm_unload_context		// 1330
 
 	.org sys_call_table + 8*NR_syscalls	// guard against failures to increase NR_syscalls
 #endif /* __IA64_ASM_PARAVIRTUALIZED_NATIVE */
--- a/arch/powerpc/include/asm/systbl.h
+++ b/arch/powerpc/include/asm/systbl.h
@@ -322,15 +322,3 @@ SYSCALL_SPU(epoll_create1)
 SYSCALL_SPU(dup3)
 SYSCALL_SPU(pipe2)
 SYSCALL(inotify_init1)
-SYSCALL(pfm_create_context)
-SYSCALL(pfm_write_pmcs)
-SYSCALL(pfm_write_pmds)
-SYSCALL(pfm_read_pmds)
-SYSCALL(pfm_load_context)
-SYSCALL(pfm_start)
-SYSCALL(pfm_stop)
-SYSCALL(pfm_restart)
-SYSCALL(pfm_create_evtsets)
-SYSCALL(pfm_getinfo_evtsets)
-SYSCALL(pfm_delete_evtsets)
-SYSCALL(pfm_unload_context)
--- a/arch/powerpc/include/asm/unistd.h
+++ b/arch/powerpc/include/asm/unistd.h
@@ -341,22 +341,10 @@
 #define __NR_dup3		316
 #define __NR_pipe2		317
 #define __NR_inotify_init1	318
-#define __NR_pfm_create_context	319
-#define __NR_pfm_write_pmcs	320
-#define __NR_pfm_write_pmds	321
-#define __NR_pfm_read_pmds	322
-#define __NR_pfm_load_context	323
-#define __NR_pfm_start		324
-#define __NR_pfm_stop		325
-#define __NR_pfm_restart	326
-#define __NR_pfm_create_evtsets	327
-#define __NR_pfm_getinfo_evtsets 328
-#define __NR_pfm_delete_evtsets 329
-#define __NR_pfm_unload_context	330
 
 #ifdef __KERNEL__
 
-#define __NR_syscalls		331
+#define __NR_syscalls		319
 
 #define __NR__exit __NR_exit
 #define NR_syscalls	__NR_syscalls
--- a/arch/x86/ia32/ia32entry.S
+++ b/arch/x86/ia32/ia32entry.S
@@ -826,16 +826,4 @@ ia32_sys_call_table:
 	.quad sys_dup3			/* 330 */
 	.quad sys_pipe2
 	.quad sys_inotify_init1
-	.quad sys_pfm_create_context
-	.quad sys_pfm_write_pmcs
-	.quad sys_pfm_write_pmds	/* 335 */
-	.quad sys_pfm_read_pmds
-	.quad sys_pfm_load_context
-	.quad sys_pfm_start
-	.quad sys_pfm_stop
-	.quad sys_pfm_restart		/* 340 */
-	.quad sys_pfm_create_evtsets
-	.quad sys_pfm_getinfo_evtsets
-	.quad sys_pfm_delete_evtsets
-	.quad sys_pfm_unload_context
 ia32_syscall_end:
--- a/arch/x86/include/asm/ia32_unistd.h
+++ b/arch/x86/include/asm/ia32_unistd.h
@@ -8,12 +8,11 @@
  * the number. This should be otherwise in sync with asm-x86/unistd_32.h. -AK
  */
 
-#define __NR_ia32_restart_syscall    0
-#define __NR_ia32_exit		     1
-#define __NR_ia32_read		     3
-#define __NR_ia32_write		     4
-#define __NR_ia32_sigreturn	     119
-#define __NR_ia32_rt_sigreturn	     173
-#define __NR_ia32_pfm_create_context 333
+#define __NR_ia32_restart_syscall 0
+#define __NR_ia32_exit		  1
+#define __NR_ia32_read		  3
+#define __NR_ia32_write		  4
+#define __NR_ia32_sigreturn	119
+#define __NR_ia32_rt_sigreturn	173
 
 #endif /* _ASM_X86_IA32_UNISTD_H */
--- a/arch/x86/include/asm/unistd_32.h
+++ b/arch/x86/include/asm/unistd_32.h
@@ -338,23 +338,9 @@
 #define __NR_dup3		330
 #define __NR_pipe2		331
 #define __NR_inotify_init1	332
-#define __NR_pfm_create_context	333
-#define __NR_pfm_write_pmcs	(__NR_pfm_create_context+1)
-#define __NR_pfm_write_pmds	(__NR_pfm_create_context+2)
-#define __NR_pfm_read_pmds	(__NR_pfm_create_context+3)
-#define __NR_pfm_load_context	(__NR_pfm_create_context+4)
-#define __NR_pfm_start		(__NR_pfm_create_context+5)
-#define __NR_pfm_stop		(__NR_pfm_create_context+6)
-#define __NR_pfm_restart	(__NR_pfm_create_context+7)
-#define __NR_pfm_create_evtsets	(__NR_pfm_create_context+8)
-#define __NR_pfm_getinfo_evtsets (__NR_pfm_create_context+9)
-#define __NR_pfm_delete_evtsets (__NR_pfm_create_context+10)
-#define __NR_pfm_unload_context	(__NR_pfm_create_context+11)
 
 #ifdef __KERNEL__
 
-#define NR_syscalls 345
-
 #define __ARCH_WANT_IPC_PARSE_VERSION
 #define __ARCH_WANT_OLD_READDIR
 #define __ARCH_WANT_OLD_STAT
--- a/arch/x86/include/asm/unistd_64.h
+++ b/arch/x86/include/asm/unistd_64.h
@@ -653,30 +653,7 @@ __SYSCALL(__NR_dup3, sys_dup3)
 __SYSCALL(__NR_pipe2, sys_pipe2)
 #define __NR_inotify_init1			294
 __SYSCALL(__NR_inotify_init1, sys_inotify_init1)
-#define __NR_pfm_create_context			295
-__SYSCALL(__NR_pfm_create_context, sys_pfm_create_context)
-#define __NR_pfm_write_pmcs			(__NR_pfm_create_context+1)
-__SYSCALL(__NR_pfm_write_pmcs, sys_pfm_write_pmcs)
-#define __NR_pfm_write_pmds			(__NR_pfm_create_context+2)
-__SYSCALL(__NR_pfm_write_pmds, sys_pfm_write_pmds)
-#define __NR_pfm_read_pmds			(__NR_pfm_create_context+3)
- __SYSCALL(__NR_pfm_read_pmds, sys_pfm_read_pmds)
-#define __NR_pfm_load_context			(__NR_pfm_create_context+4)
-__SYSCALL(__NR_pfm_load_context, sys_pfm_load_context)
-#define __NR_pfm_start				(__NR_pfm_create_context+5)
-__SYSCALL(__NR_pfm_start, sys_pfm_start)
-#define __NR_pfm_stop				(__NR_pfm_create_context+6)
-__SYSCALL(__NR_pfm_stop, sys_pfm_stop)
-#define __NR_pfm_restart			(__NR_pfm_create_context+7)
-__SYSCALL(__NR_pfm_restart, sys_pfm_restart)
-#define __NR_pfm_create_evtsets			(__NR_pfm_create_context+8)
-__SYSCALL(__NR_pfm_create_evtsets, sys_pfm_create_evtsets)
-#define __NR_pfm_getinfo_evtsets 		(__NR_pfm_create_context+9)
-__SYSCALL(__NR_pfm_getinfo_evtsets, sys_pfm_getinfo_evtsets)
-#define __NR_pfm_delete_evtsets 		(__NR_pfm_create_context+10)
-__SYSCALL(__NR_pfm_delete_evtsets, sys_pfm_delete_evtsets)
-#define __NR_pfm_unload_context			(__NR_pfm_create_context+11)
-__SYSCALL(__NR_pfm_unload_context, sys_pfm_unload_context)
+
 
 #ifndef __NO_STUBS
 #define __ARCH_WANT_OLD_READDIR
--- a/arch/x86/kernel/syscall_table_32.S
+++ b/arch/x86/kernel/syscall_table_32.S
@@ -332,15 +332,3 @@ ENTRY(sys_call_table)
 	.long sys_dup3			/* 330 */
 	.long sys_pipe2
 	.long sys_inotify_init1
-	.long sys_pfm_create_context
-	.long sys_pfm_write_pmcs
-	.long sys_pfm_write_pmds	/* 335 */
-	.long sys_pfm_read_pmds
-	.long sys_pfm_load_context
-	.long sys_pfm_start
-	.long sys_pfm_stop
-	.long sys_pfm_restart		/* 340 */
-	.long sys_pfm_create_evtsets
-	.long sys_pfm_getinfo_evtsets
-	.long sys_pfm_delete_evtsets
-	.long sys_pfm_unload_context
--- a/kernel/sys_ni.c
+++ b/kernel/sys_ni.c
@@ -133,19 +133,6 @@ cond_syscall(sys_io_cancel);
 cond_syscall(sys_io_getevents);
 cond_syscall(sys_syslog);
 
-cond_syscall(sys_pfm_create_context);
-cond_syscall(sys_pfm_write_pmcs);
-cond_syscall(sys_pfm_write_pmds);
-cond_syscall(sys_pfm_read_pmds);
-cond_syscall(sys_pfm_restart);
-cond_syscall(sys_pfm_start);
-cond_syscall(sys_pfm_stop);
-cond_syscall(sys_pfm_load_context);
-cond_syscall(sys_pfm_unload_context);
-cond_syscall(sys_pfm_create_evtsets);
-cond_syscall(sys_pfm_delete_evtsets);
-cond_syscall(sys_pfm_getinfo_evtsets);
-
 /* arch-specific weak syscall entries */
 cond_syscall(sys_pciconfig_read);
 cond_syscall(sys_pciconfig_write);
